### DISCUSSION TITLE
Support for Remote Tunnels

### DISCUSSION THREAD

**Discussion by const-ae** [posit-dev/positron#4944] [Q&A]
VSCode supports two different ways to run code on a remote server: via SSH connections or [tunnels](https://code.visualstudio.com/docs/remote/tunnels). There is already quite a bit of effort to enable SSH connections using the Open Remote SSH extension. Are there also plans to support tunnels?

I am interested in this feature, as I am on a university-managed cluster system, where we use Sun Grid Engine to submit jobs. VSCode's Tunnels allow me to submit an interactive job and launch a VSCode instance inside so that the cluster can manage the resources. In contrast, with the SSH extension, I can only connect to the head node, where heavy computations are forbidden.

**Comment by jmcphers** [comment]
Not at present, though feel free to file a feature request. A major challenge with supporting dev tunnels is that they rely on a proprietary Microsoft cloud relay service for handshake/discovery and auth, which we would need to rebuild or license for use in Positron. 

One way you could achieve this with existing technology would be to use a reverse SSH tunnel at a designated port on the head node, which would enable Positron to connect through the head node as a gateway.

```mermaid
graph LR
positron -- ssh:2345 --> hn[head node]
cl[cluster node] -- reverse ssh --> hn
hn -- ssh:22 --> cl
```

**Comment by matiasandina** [comment]
Is there any plan / update on this end? It would be good to be able to have an equivalent of tunnel with github handling the auth instead of ssh. Even if it's a single connection, university IT firewall rules can easily lock you out of access when moving from building to building. Tunnels as implemented in VSCode can by-pass that and allow remote connections to a running server.
